### PR TITLE
Update requirements files

### PR DESCRIPTION
**Pull request by jcristau**
This:
- fixes our requirements file to include six, which was missing
- replaces the update-requirements.sh script with a pin.sh script using docker, for more predictable output and ease of running locally
- bumps dependencies using the new script

### CODE REVIEW

File: maintenance/pin-helper.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# This runs in docker to pin our requirements files
+
+set -e
+
+pip install pip-tools
+
+rm -f *requirements${SUFFIX}.txt
+pip-compile --generate-hashes --output-file requirements${SUFFIX}.txt requirements.in
+pip-compile --generate-hashes --output-file test-requirements${SUFFIX}.txt test-requirements.in
```
Comment (JohanLorenzo) [inline maintenance/pin-helper.sh:10]: Nit: let's prevent unwanted shell injections by safe-guarding this variable

```suggestion
rm -f *"requirements${SUFFIX}.txt"
pip-compile --generate-hashes --output-file "requirements${SUFFIX}.txt" requirements.in
pip-compile --generate-hashes --output-file "test-requirements${SUFFIX}.txt" test-requirements.in
```

File: maintenance/pin.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -e
+set -x
+
+docker run -t -v $PWD:/src -w /src -e SUFFIX=-py2 python:2.7 maintenance/pin-helper.sh "$@"
+docker run -t -v $PWD:/src -w /src python:3.6 maintenance/pin-helper.sh "$@"
```
Comment (JohanLorenzo) [inline maintenance/pin.sh:7]: Question: what is the use case for using "$@" here?

**Review by JohanLorenzo — APPROVED** [review]
LGTM modulo a small nit!

File: maintenance/pin.sh (hunk 2 above)
Comment (jcristau) [reply maintenance/pin.sh:7]: There probably isn't one. :)

**Pull request merged**